File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        j = 0
        res = sorted(nums)
        idx = bisect.bisect_right(res, pivot)
        for i in range(len(nums)):
            if nums[i] < pivot:
                res[j] = nums[i]
                j += 1
            elif nums[i] > pivot:
                res[idx] = nums[i]
                idx += 1
        return res

File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        int j = 0;
        vector<int> res(nums);
        sort(res.begin(),res.end());
        int idx = upper_bound(res.begin(), res.end(), pivot) - res.begin();
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] < pivot) {
                res[j] = nums[i];
                j++;
            } else if (nums[i] > pivot) {
                res[idx] = nums[i];
                idx++;
            }
        }
        return res;
    }
};